File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.IO;
using System.Windows.Forms;
using DevExpress.Drawing;
using DevExpress.XtraCharts;

namespace ImageLegendItems {

    public partial class Form1 : Form {
        Dictionary<string, DXImage> photos = new Dictionary<string, DXImage>();

        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            this.employeesTableAdapter.Fill(this.nwindDataSet.Employees);

            InitializePhotos();
        }

        void InitializePhotos() {
            foreach (DataRow row in nwindDataSet.Employees.Rows) {
                string lastName = row.ItemArray[1].ToString();
                if (!photos.ContainsKey(lastName)) {
                    using (MemoryStream stream = new MemoryStream((byte[])row.ItemArray[14])) {
                        using (DXImage sourceImage = DXImage.FromStream(stream)) {
                            DXBitmap image = new DXBitmap(74, 79);
                            using (DXGraphics graphics = DXGraphics.FromImage(image))
                                graphics.DrawImage(sourceImage, 
                                    new Rectangle(new Point(5, 5), new Size(75, 75)));
                            photos.Add(lastName, image);
                        }
                    }
                }
            }
        }

        private void chartControl1_CustomDrawSeries(object sender, CustomDrawSeriesEventArgs e) {
            e.DXLegendMarkerImage = photos[e.Series.Name];
        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Drawing
Imports System.IO
Imports System.Windows.Forms
Imports DevExpress.XtraCharts

Namespace ImageLegendItems

    Public Partial Class Form1
        Inherits Form

        Private photos As Dictionary(Of String, Image) = New Dictionary(Of String, Image)()

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            employeesTableAdapter.Fill(nwindDataSet.Employees)
            InitializePhotos()
        End Sub

        Private Sub InitializePhotos()
            For Each row As DataRow In nwindDataSet.Employees.Rows
                Dim lastName As String = row.ItemArray(1).ToString()
                If Not photos.ContainsKey(lastName) Then
                    Using stream As MemoryStream = New MemoryStream(CType(row.ItemArray(14), Byte()))
                        Using sourceImage As Image = Image.FromStream(stream)
                            Dim image As Bitmap = New Bitmap(74, 79)
                            Using graphics As Graphics = Graphics.FromImage(image)
                                graphics.DrawImage(sourceImage, New Rectangle(New Point(5, 5), New Size(75, 75)))
                            End Using

                            photos.Add(lastName, image)
                        End Using
                    End Using
                End If
            Next
        End Sub

        Private Sub chartControl1_CustomDrawSeries(ByVal sender As Object, ByVal e As CustomDrawSeriesEventArgs)
            e.LegendMarkerImage = photos(e.Series.Name)
        End Sub
    End Class
End Namespace
